### PR TITLE
Update `OffsiteLink` url validation

### PR DESCRIPTION
**Pull request by gpeng**
Currently the validation for `OffsiteLink#url` only allows `gov.uk` hosts. This PR also allows:

```
gov.wales
tse-lab-net.eu
flu-lab-net.eu
```

[Trello](https://trello.com/c/X5Ur3A5p/375-offsite-links-function-in-whitehall-not-allowing-offsite-links-small)

Also included in this PR are copy updates to more accurately describe the 'Offsite Link'.

Before:

window_and_featured_items_on_global_summit_to_end_sexual_violence_in_conflict_-_admin_-_gov_uk

window_and_create_an_offsite_link_-_admin_-_gov_uk

After: 

window_and_featured_items_on_global_summit_to_end_sexual_violence_in_conflict_-_admin_-_gov_uk

window_and_create_an_offsite_link_-_admin_-_gov_uk

window_and_create_an_offsite_link_-_admin_-_gov_uk

### CODE REVIEW

File: app/models/offsite_link.rb [hunk 1]
```diff
@@ -26,4 +25,29 @@ def humanized_link_type
...
+    host =~ /gov\.uk$/
+  end
+
+  def url_is_whitelisted?(host)
+    whitelisted_hosts = [
+      "flu-lab-net.eu",
+      "tse-lab-net.eu",
+    ]
+
+    whitelisted_hosts.each do |whitelisted_host|
```
Comment (boffbowsh) [inline app/models/offsite_link.rb]:
```ruby
whitelisted_hosts.any? { |wl_host| host =~ /#wl_host}$/ }
```

**Comment by boffbowsh** [comment]
Tiny comment otherwise 👍

File: app/models/offsite_link.rb (hunk 1 above)
Comment (gpeng) [inline app/models/offsite_link.rb]: Fixed in 25729d9. Also slimmed the preceding methods down a bit. The `nil` check is redundant now as `nil =~` is valid.

**Comment by boffbowsh** [comment]
👍

**Comment by gpeng** [comment]
Rebased to fix a few features broken by the copy changes. Merging...

**Pull request merged**